Observer/NonBoltOrderObserver.php

```diff
@@ -208,6 +217,84 @@ protected function getPhone($cart)
         return $cart['shipments'][0]['shipping_address']['phone'];
     }
 
+    /**
+     * @param array $cart
+     * @param CustomerInterface $customer
+     * @return string
+     */
+    protected function getEmail($cart, $customer)
+    {
+        if (!is_null($customer->getEmail())) {
+            return $customer->getEmail();
+        }
+
+        if (!@$cart['shipments']) {
```

Comment:
I think the most native PHP way for this
```
if (isset($cart['shipments'][0]['shipping_address'])) {
 return $cart['shipments'][0]['shipping_address']['email'];
}
```
This code works ok even if $cart['shipments'] undefined

Reply:
Are you sure? I just tried this and received an error:
```
Notice: Undefined index: shipments in /Applications/MAMP/htdocs/magento2/vendor/boltpay/bolt-magento2/Observer/NonBoltOrderObserver.php:206.
```

Reply:
wait never mind I wrote the test wrong